cmd/main.go

```diff
@@ -167,26 +167,45 @@ func main() {
 func leaseHandler(podName string, mc controller.ModifyController, leaseChannel chan *v1.Lease) {
 	var cancel context.CancelFunc = nil
 
-	for lease := range leaseChannel {
-		currentLeader := *lease.Spec.HolderIdentity
-
-		klog.V(6).InfoS("leaseHandler: Lease updated", "currentLeader", currentLeader, "podName", podName)
-
-		if currentLeader == podName && cancel == nil {
-			var ctx context.Context
-			ctx, cancel = context.WithCancel(context.Background())
-			klog.InfoS("leaseHandler: Starting ModifyController", "podName", podName, "currentLeader", currentLeader)
-			go mc.Run(*workers, ctx)
-		} else if currentLeader != podName && cancel != nil {
-			klog.InfoS("leaseHandler: Stopping ModifyController", "podName", podName, "currentLeader", currentLeader)
-			cancel()
-			cancel = nil
-		}
-	}
+	klog.InfoS("leaseHandler: Looking for external-resizer lease holder")
+
+	timer := time.NewTimer(*resyncPeriod)
+	defer timer.Stop()
+
+	for {
+		select {
+		case lease, ok := <-leaseChannel:
+			if !ok {
+				if cancel != nil {
+					cancel()
+				}
+				return
+			}
+			currentLeader := *lease.Spec.HolderIdentity
+			klog.V(6).InfoS("leaseHandler: Lease updated", "currentLeader", currentLeader, "podName", podName)
+
+			if currentLeader == podName && cancel == nil {
+				var ctx context.Context
+				ctx, cancel = context.WithCancel(context.Background())
+				klog.InfoS("leaseHandler: Starting ModifyController", "podName", podName, "currentLeader", currentLeader)
+				go mc.Run(*workers, ctx)
+			} else if currentLeader != podName && cancel != nil {
+				klog.InfoS("leaseHandler: Stopping ModifyController", "podName", podName, "currentLeader", currentLeader)
+				cancel()
+				cancel = nil
+			}
+
+			if !timer.Stop() {
+				<-timer.C
+			}
+			timer.Reset(*resyncPeriod)
 
-	// Ensure cancel is called if it's not nil when we exit the function
-	if cancel != nil {
-		cancel()
+		case <-timer.C:
+			if cancel != nil {
+				cancel()
+			}
+			klog.Fatalf("leaseHandler: No external-resizer lease update received within timeout period", "timeout", *resyncPeriod)
```

Comment:
Following up on the offline convo between you and Wayne:

I personally would not find it obvious that the root cause of "No external-resizer lease update received within timeout period" would be external-resizer leader election being disabled. 

As a customer, I would assume that this sidecar has its own leader election implementation and doesn't rely on the `external-resizer`. Especially because there is no mention of this design decision except in your newly added requirements text. 

I would either explicitly call this root cause out in a `troubleshooting` document in this repo's README.md or [the aws-ebs-csi-driver FAQ](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/docs/faq.md). (To distinguish it from some kind of rare distributed systems error)

Reply:
> Especially because there is no mention of this design decision except in your newly added requirements text.

This is a good point, added an additional log statement to list the requirement in the container logs.

> Therefore I would either explicitly call this root cause out in a troubleshooting document in this repo's README.md or [the aws-ebs-csi-driver FAQ](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/docs/faq.md)

Thinking ahead, I think it would be best not to make assumptions about the root cause - even if it is correct in the majority of cases. The improved logging should help steer cluster operators towards the root cause without providing the opportunity for a misdiagnosis (think of scenarios where leader election is enabled but no lease update is received within the timeout period due to bugs in the leases code or (unlikely scenario) lease name is changed in the resizer, etc).